wazo_confd/plugins/user/middleware.py

```diff
@@ -396,3 +397,17 @@ def update(self, user_id, body, tenant_uuid, tenant_uuids, recursive=False):
                     tenant_uuid,
                     tenant_uuids,
                 )
+
+            if agent:
+                agent_id = user.agentid
+                self._middleware_handle.get('user_agent_association').dissociate(
+                    user_id, tenant_uuids
+                )
+                self._middleware_handle.get('agent').delete(agent_id, tenant_uuids)
```

Comment:
Doing a delete/create will generate a new agent id. This would probably break some stuff. The first thing that comes to mind is the agent statistics which are based on the agent's ID. Doing a delete/create would either leave some orphaned agent statistics or delete them, both are bad.